flink-architecture-tests/flink-architecture-tests-test/README.md

```diff
@@ -0,0 +1,55 @@
+# flink-architecture-tests-test
+
+This submodule contains rules defined centrally for test code architectural tests that will be
+developed in each submodule individually where the test code has been developed.
+
+## Why the architecture test infra for production code and for test code are different
+
+Comparing to central production code architectural tests, test code architectural tests chose this
+central rules plus distributed tests for the following reasons:
+
+- Reduce the classpath complexity. Developing test code architectural tests centrally will need the
+  classpath with dependencies to be exactly same in different runtime environments, i.e. maven and
+  IntelliJ IDEA. This turns out that, on one side, every Flink submodule have to create test jar,
+  and on the other hand, the test jar must contain all test classes. In the current, there are
+  already some Flink submodules that created test jar with individual exclusion filter logic which
+  has conflict with the second prerequisite. Creating more test jars increase the complexity of
+  dependencies and have impact on the project stability.
+- Separation of concern. Developing the test code architectural tests within the submodule will keep
+  the focus on the submodule's test code. Different from production code test, there are less code
+  sharing for the test between Flink submodules, therefore less performance issue to be considered.
+- DYI and SRP. There are commons when testing the architecture of the test code. These commons can
+  be defined as common rules centrally in this submodule and used by each individual architectural
+  test in their own submodules. This module will take the responsibility for central rules while
+  individual test in each submodule will take the responsibility for the local architectural tests.
+- Flexibility. For special cases, further rules could be defined locally in their own submodule and
+  used locally.
```

Comment:
```suggestion
Compared to `flink-architecture-tests-production` which implements and executes architectural rules centrally, `flink-architecture-tests-tests` implements the rules centrally, but they are executed individually in each submodule. This is done for the following reasons:

- Reduce the classpath complexity. In order to execute the tests centrally, each submodule would have to provide a test-jar. However, this does not work well in IntelliJ which doesn't support all Maven configurations (such as exclusion filters); this would cause the rules to behave differently when executed from Maven versus IntelliJ. Furthermore, creating more test-jars is not desirable and increases the project complexity.
- Separation of concerns. Test code should be viewed as internal to a module, and by separating the violation stores for each submodule this separation can be maintained. Unlike the production code tests there is little shared code between test code, and thus the performance benefits do not apply.
- Flexibility. Each submodule can import the generic rules defined in `flink-architecture-tests-tests`, but also add further module-specific architectural tests.
```